### PR TITLE
feat(TBD-15770): support insert overwrite ticebergOutput

### PR DESCRIPTION
**Pull request by zshen-talend**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_java.xml [hunk 1]
```diff
@@ -25,12 +25,27 @@
...
+		<PARAMETER NAME="USE_BATCH" FIELD="CHECK" NUM_ROW="13" SHOW_IF="(USE_INSERT_OVERWRITE=='false')">
 			<DEFAULT>false</DEFAULT>
 		</PARAMETER>
 		<PARAMETER NAME="BATCH_SIZE" FIELD="TEXT" SHOW_IF="USE_BATCH=='true'" NUM_ROW="14">
 			<DEFAULT>10</DEFAULT>
 		</PARAMETER>
+		<PARAMETER NAME="USE_INSERT_OVERWRITE" FIELD="CHECK" NUM_ROW="15" SHOW_IF="(USE_BATCH=='false')">
+			<DEFAULT>false</DEFAULT>
+		</PARAMETER>
+		<PARAMETER NAME="All_ROW_FROM_SOURCE_TABLE" FIELD="RADIO" REQUIRED="true" NUM_ROW="16" GROUP="MODE" SHOW_IF="USE_INSERT_OVERWRITE=='true'">
```
Comment (lbourgeois) [inline main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_java.xml:37]:
```suggestion
		<PARAMETER NAME="ALL_ROWS_FROM_SOURCE_TABLE" FIELD="RADIO" REQUIRED="true" NUM_ROW="16" GROUP="OVERWRITE_SOURCE" SHOW_IF="USE_INSERT_OVERWRITE=='true'">
```

File: main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_messages.properties [hunk 2]
```diff
@@ -3,4 +3,10 @@ LONG_NAME=Writes data to Iceberg
 BATCH_SIZE.NAME=Batch size
 CONNECTION.NAME=Connection
 TABLE.NAME=Table
-USE_BATCH.NAME=Use batch
\ No newline at end of file
+USE_BATCH.NAME=Use batch
+USE_INSERT_OVERWRITE.NAME=Use insert overwrite
+All_ROW_FROM_SOURCE_TABLE.NAME=All row from source table
```
Comment (lbourgeois) [inline main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_messages.properties:8]:
```suggestion
ALL_ROWS_FROM_SOURCE_TABLE.NAME=All rows from source table
```

File: main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_main.javajet [hunk 3]
```diff
@@ -9,10 +9,15 @@ imports="
 final TIcebergOutput ticebergOutput = new TIcebergOutput((CodeGeneratorArgument) argument);
 %>
 
-org.talend.bigdata.core.di.components.ImmutableRecord record_<%=ticebergOutput.getCid()%> = org.talend.bigdata.core.di.components.ImmutableRecord.builder()
-                    <% for (org.talend.designer.bigdata.di.components.Schema.Field field : ticebergOutput.getFields()) {%>
-                        .putValues("<%=field.getName()%>", <%=ticebergOutput.getInputVariable()%>.<%=field.getName()%>)
-                    <%}%>
-                        .build();
+<%if(ticebergOutput.isMainFlowExist()&&!ticebergOutput.useInsertOverwrite()){%>
```
Comment (lbourgeois) [inline main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_main.javajet:12]: Using ticebergOutput.isMainFlowExist() seems useless here.
Testing !ticebergOutput.useInsertOverwrite() seems enough (like it is done in the _begin.javajet)

Comment (zshen-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_main.javajet:12]: It is work well after remove it. So modify as request

File: main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_messages.properties (hunk 2 above)
Comment (zshen-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_messages.properties]: modify it

File: main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_java.xml (hunk 1 above)
Comment (zshen-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_java.xml]: modify it

File: main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_java.xml [hunk 4]
```diff
@@ -25,12 +25,27 @@
...
+		<PARAMETER NAME="USE_BATCH" FIELD="CHECK" NUM_ROW="13" SHOW_IF="(USE_INSERT_OVERWRITE=='false')">
 			<DEFAULT>false</DEFAULT>
 		</PARAMETER>
 		<PARAMETER NAME="BATCH_SIZE" FIELD="TEXT" SHOW_IF="USE_BATCH=='true'" NUM_ROW="14">
 			<DEFAULT>10</DEFAULT>
 		</PARAMETER>
+		<PARAMETER NAME="USE_INSERT_OVERWRITE" FIELD="CHECK" NUM_ROW="15" SHOW_IF="(USE_BATCH=='false')">
+			<DEFAULT>false</DEFAULT>
+		</PARAMETER>
+		<PARAMETER NAME="All_ROWS_FROM_SOURCE_TABLE" FIELD="RADIO" REQUIRED="true" NUM_ROW="16" GROUP="OVERWRITE_SOURCE" SHOW_IF="USE_INSERT_OVERWRITE=='true'">
```
Comment (lbourgeois) [inline main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_java.xml:37]:
```suggestion
		<PARAMETER NAME="ALL_ROWS_FROM_SOURCE_TABLE" FIELD="RADIO" REQUIRED="true" NUM_ROW="16" GROUP="OVERWRITE_SOURCE" SHOW_IF="USE_INSERT_OVERWRITE=='true'">
```
to put it in full capital letters please

File: main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_main.javajet (hunk 3 above)
Comment (lbourgeois) [reply main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_main.javajet:12]: Ok got that. Can you renaming isMainFlowExist() to hasMainFlow() would be better I think

File: main/plugins/org.talend.designer.bigdata.di/src/main/java/org/talend/designer/bigdata/di/components/iceberg/TIcebergOutput.java [hunk 5]
```diff
@@ -1,12 +1,22 @@
...
+import org.talend.core.model.process.EConnectionType;
+import org.talend.core.model.process.INode;
 import org.talend.designer.bigdata.di.components.DesignerDIComponent;
 import org.talend.designer.codegen.config.CodeGeneratorArgument;
 
 public class TIcebergOutput implements DesignerDIComponent.WithOneInput, DesignerDIComponent.WithSchema, DesignerDIComponent.WithConnection {
 
     protected final CodeGeneratorArgument codeGeneratorArgument;
 
+    public boolean isMainFlowExist() {
```
Comment (lbourgeois) [inline main/plugins/org.talend.designer.bigdata.di/src/main/java/org/talend/designer/bigdata/di/components/iceberg/TIcebergOutput.java:12]:
```suggestion
    public boolean hasMainFlow() {
```

Comment (zshen-talend) [reply main/plugins/org.talend.designer.bigdata.di/src/main/java/org/talend/designer/bigdata/di/components/iceberg/TIcebergOutput.java:12]: done it

File: main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_java.xml (hunk 4 above)
Comment (zshen-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tIcebergOutput/tIcebergOutput_java.xml:37]: modify it

**Review by lbourgeois — DISMISSED** [review]

**Review by lbourgeois — APPROVED** [review]

**Pull request merged**